src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -53,6 +53,12 @@ internal static object Create(Type type)
                 case TypeCode.String:
                     return new JsonValueConverterString();
             }
+
+            if (type == typeof(DateTimeOffset))
+            {
+                return new JsonValueConverterDateTimeOffset();
```

Comment:
Should this be part of the type code enum and the above switch statement?

Reply:
It's type code `object`. Given offline feedback from @steveharter, this placement should be fine.

Reply:
That seems odd to me when we have `TypeCode` for `DateTime`. I think adding `DateTimeOffset` like this, seemingly as a "custom" type is strange, but maybe that's fine if the serializer/deserializer treats `DateTimeOffset` differently than the rest.

Reply:
So, after talking to @steveharter, and since we we use `IConvertible` to detect "primitive value", `DateTimeOffset`/`Guid`/etc would need to be special cased:
https://github.com/dotnet/corefx/blob/b0751dcd4a419ba6731dcaa7d240a8a1946c934c/src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs#L228-L231